syncstorage/db.go

```diff
@@ -217,23 +219,22 @@ func NewDB(path string, conf *Config) (*DB, error) {
   work is handled by private functions.
 */
 
-// LastModified gets the database modified time
-func (d *DB) LastModified() (modified int, err error) {
+// LastModified returns the top level last modified timestamp
+func (d *DB) LastModified() (int, error) {
 	d.Lock()
 	defer d.Unlock()
 
-	var m sql.NullInt64
+	lastMod, err := getKey(d.db, STORAGE_LAST_MODIFIED)
+	if lastMod == "" || err != nil {
+		return 0, err
```

Comment:
In the current implementation, we don't actually look at x-last-modified of the response itself. We take the last-modified from info/collections, then advance it if the *records* in a get response have a later timestamp. Finally, we use whatever value we end up with to make the POST.

IOW, I expect that if get returns no records, we will do a POST with x-if-unmodified-since of whatever info/collections said the timestamp was.